idxmgmt/supporter_factory.go

```diff
@@ -0,0 +1,93 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package idxmgmt
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/idxmgmt"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/template"
+)
+
+// functionality largely copied from libbeat
+
+// MakeDefaultSupporter creates the index management supporter for APM that is passed to libbeat.
+func MakeDefaultSupporter(log *logp.Logger, info beat.Info, configRoot *common.Config) (idxmgmt.Supporter, error) {
+
+	const logName = "index-management"
+
+	cfg := struct {
+		ILM       *common.Config         `config:"setup.ilm"`
+		Template  *common.Config         `config:"setup.template"`
+		Output    common.ConfigNamespace `config:"output"`
+		Migration *common.Config         `config:"migration"`
+	}{}
+	if configRoot != nil {
+		if err := configRoot.Unpack(&cfg); err != nil {
+			return nil, err
+		}
+	}
+
+	if log == nil {
+		log = logp.NewLogger(logName)
+	} else {
+		log = log.Named(logName)
+	}
+
+	if err := checkTemplateESSettings(cfg.Template, cfg.Output); err != nil {
+		return nil, err
+	}
+
+	return newSupporter(log, info, cfg.Template, cfg.ILM, cfg.Migration.Enabled())
+}
+
+func checkTemplateESSettings(tmpl *common.Config, out common.ConfigNamespace) error {
```

Comment:
As silly as it sounds, I generally have a better time reading full words even if they are long (`indexTemplate`, `indexManagement`) rather than made up abbreviations (`templ`, `indxmngmt`, etc), with some exception (`config` is ok, but `cfg` is harder already). 
It is also easier if variable names in argument declaration are the same as the variable names in calling functions (`template`, `output`).
This is not important, and I am aware that happens all over the place. It is just that i am reading a lot of code lately :)